autoload/clap/common_history.vim

```diff
@@ -0,0 +1,33 @@
+" Author: liuchengxu <xuliuchengxlc@gmail.com>
+" Description: Common utilities for command/search history.
+
+let s:save_cpo = &cpoptions
+set cpoptions&vim
+
+" Derived from fzf.vim
+function! s:get_history_list(type) abort
+  let max  = histnr(a:type)
+  let fmt  = ' %'.len(string(max)).'d '
+  let list = filter(map(range(1, max), 'histget("'. a:type .'", - v:val)'), '!empty(v:val)')
+  return list
+endfunction
+
+function! clap#common_history#source(type) abort
+  let hist_list = s:get_history_list(a:type)
+  let hist_len = len(hist_list)
+  return map(hist_list, 'printf("%4d", hist_len - v:key)."  ".v:val')
+endfunction
+
+function! clap#common_history#sink(type, selected) abort
+  let item = matchstr(a:selected, '\d\+\s\+\zs\(.*\)')
+  if a:type ==# ':'
+    call histadd(':', item)
+    call execute(item, '')
+  elseif a:type ==# '/'
```

Comment:
Have you tested yourself?  `/` sink does not always work as expected, the editor can be freezed and I have to press `<C-C>` to exit.

Reply:
Yes, I tested. I never meet this issue before. Can you reproduce it?

Reply:
My issue is due to `n` is mapped in my config.

```diff
diff --git a/autoload/clap/common_history.vim b/autoload/clap/common_history.vim
index 444eb11..0efbc8b 100644
--- a/autoload/clap/common_history.vim
+++ b/autoload/clap/common_history.vim
@@ -25,7 +25,7 @@ function! clap#common_history#sink(type, selected) abort
     call execute(item, '')
   elseif a:type ==# '/'
     let @/ = item
-    execute 'normal n'
+    normal! n
   endif
 endfunction
```

Using `normal!` fixes my issue, a plugin should usually should use `normal!` for avoiding such issues.

Reply:
Ah, okay!

Reply:
Done!